shanoir-ng-front/src/app/studies/study-list/study-list.component.ts

```diff
@@ -96,33 +99,41 @@ export class StudyListComponent extends BrowserPaginEntityListComponent<Study> {
                     }
                 }
             }
-            let ids = new Set<number>();
-            studies.forEach(study => {
-               ids.add(study.id);
-            });
-            this.studyService.getStudiesStorageVolume(ids).then(volumes => {
-                studies.forEach( study => {
-                    (study as Study).totalSize = volumes.get(study.id)?.total;
-                    let sizesByLabel = new Map<String, number>()
-
-                    if(volumes.get(study.id)?.volumeByFormat){
-                        for(let sizeByFormat of volumes.get(study.id)?.volumeByFormat){
-                            if(sizeByFormat.size > 0){
-                                sizesByLabel.set(DatasetExpressionFormat.getLabel(sizeByFormat.format), sizeByFormat.size);
+            this.fetchStorageVolumes(studies);
+        });
+        return earlyResult;
+    }
+
+    private fetchStorageVolumes(studies: Study[] | AccessRequest[]) {
+        let pageSize = Number(this.table.maxResults);
+        let promises = [];
+        for (let i = 0; i < studies.length; i += pageSize) {
```

Comment:
Why do we do it page by page ? Is it too long otherwise ?

Reply:
It is quite weird that we can start to sort by size wereas all study size are not loaded yet -> The display can be then misleading

Reply:
Hi! I agree with Jean-Come. Users will find our tool strange, when we propose a filter on size, but only consider the current page.